### PR TITLE
Hw1

### PR DESCRIPTION
**Pull request by Genateep**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -7,16 +7,16 @@ on: [push]
...
       - name: Run script
         run: |
-          pip install black isort pytest
+          pip install flake8 isort pytest
           ls -la
-          black .
+          flake8 .
           isort --check .
-          py.test tests
+          py.test homework1/tests
```
Comment (icanhardcode) [inline .github/workflows/main.yml]: are u going to add rows like that for others HWs? return previous version pls

File: homework1/task02.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+from typing import Sequence
+
+
+def check_fib(data: Sequence[int]) -> bool:
```
Comment (icanhardcode) [inline homework1/task02.py:4]: let it be the fibb sequence if length of data >= 3

File: homework1/task03.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+def find_maximum_and_minimum(file_name: str) -> Tuple[int, int]:
+
+    """Returns a tuple with the max and min values"""
+
+    nums = set()
+
+    with open(file_name) as fi:
+        for line in fi:
+            if len(line.strip()) > 0:
+                nums.update(map(int, line.split()))
```
Comment (icanhardcode) [inline homework1/task03.py]: ```set.add()```

File: homework1/task03.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
+from typing import Tuple
+
+
+def find_maximum_and_minimum(file_name: str) -> Tuple[int, int]:
+
+    """Returns a tuple with the max and min values"""
```
Comment (icanhardcode) [inline homework1/task03.py:6]: check the docstring and func's name.. and watch what u return

File: homework1/tests/test_task02.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
+from homework1.task02 import check_fib
+
```
Comment (icanhardcode) [inline homework1/tests/test_task02.py]: try to think.. are there any corner cases?

File: homework1/tests/test_task03.py [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import pytest
+
+from homework1.task03 import find_maximum_and_minimum
+
+
+@pytest.mark.parametrize(
```
Comment (icanhardcode) [inline homework1/tests/test_task03.py]: u dont need parametrize if u have only one test

File: homework1/tests/test_task05.py [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import pytest
+
+from homework1.task05 import find_maximal_subarray_sum
+
+
```
Comment (icanhardcode) [inline homework1/tests/test_task05.py]: negative cases?

**Review by icanhardcode — CHANGES_REQUESTED** [review]

File: homework1/task03.py (hunk 4 above)
Comment (Genateep) [reply homework1/task03.py:6]: "For example for [1, 2, 3, 4, 5], function should return [1, 5]"  I took this example to work. But yes, it contradicts the title for me. Swapped the values

File: homework1/tests/test_task02.py (hunk 5 above)
Comment (Genateep) [reply homework1/tests/test_task02.py]: tried... could you throw a hint?

**Review by icanhardcode — COMMENTED** [review]
corner cases for fibbonachi task: hint- [1] - is it a fibb seq?
